src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -52,69 +52,427 @@ public class RepoConfiguration {
-
-        if (!organization.isEmpty()) {
-            repoFolderName = organization + "_" + repoFolderName;
-            displayName = organization + "/" + displayName;
-            outputFolderName = organization + "_" + outputFolderName;
+    /**
+     * Processes the organization name of the repository and updates the relevant
+     * names of the repository configs.
+     */
+    private void processOrganization(RepoBuilder repoBuilder) {
+        String org = location.getOrganization();
+
+        if (!org.isEmpty()) {
+            this.repoFolderName = repoBuilder.repoFolderName == null
+                    ? org + "_" + this.repoFolderName
+                    : repoBuilder.repoFolderName;
+            this.displayName = repoBuilder.displayName == null
+                    ? org + "/" + this.displayName
+                    : repoBuilder.displayName;
+            this.outputFolderName = repoBuilder.outputFolderName == null
+                    ? org + "_" + this.outputFolderName
+                    : repoBuilder.outputFolderName;
+        }
+    }
+
+    /**
+     * Builds the necessary configurations for RepoConfiguration.
+     * Obeys the Builder pattern as described in {@link CliArguments}.
+     */
+    public static class RepoBuilder {
+        private RepoLocation location;
+        private String branch = DEFAULT_BRANCH;
+        private String displayName;
+        private String outputFolderName;
+        private transient String extraOutputFolderName = DEFAULT_EXTRA_OUTPUT_FOLDER_NAME;
+        private transient ZoneId zoneId;
+        private transient LocalDateTime sinceDate;
+        private transient LocalDateTime untilDate;
+        private transient String repoFolderName;
+
+        private transient FileTypeManager fileTypeManager = new FileTypeManager(Collections.emptyList());
+        private transient List<String> ignoreGlobList = new ArrayList<>();
+        private transient List<String> ignoredAuthorsList = new ArrayList<>();
+        private transient AuthorConfiguration authorConfig;
+        private transient boolean isStandaloneConfigIgnored = false;
+        private transient boolean isFileSizeLimitIgnored = false;
+        private transient List<CommitHash> ignoreCommitList = Collections.emptyList();
+        private transient boolean isLastModifiedDateIncluded;
+        private transient boolean isShallowCloningPerformed = false;
+        private transient boolean isFindingPreviousAuthorsPerformed = false;
+        private transient boolean isFormatsOverriding = false;
+        private transient boolean isIgnoreGlobListOverriding;
+        private transient boolean isIgnoreCommitListOverriding = false;
+        private transient boolean isIgnoredAuthorsListOverriding;
+        private transient long fileSizeLimit = DEFAULT_FILE_SIZE_LIMIT;
+        private transient boolean isFileSizeLimitOverriding = false;
+        private transient boolean isIgnoredFileAnalysisSkipped = false;
```

Comment:
for discussion: since the builder is so tightly coupled with `RepoConfiguration`, would it make sense for the builder to hold a `RepoConfiguration` and build directly into it? This would save us the duplicated fields and boilerplate copies that seem error-prone. 

I think the Oracle [blog post](https://blogs.oracle.com/javamagazine/post/exploring-joshua-blochs-builder-design-pattern-in-java#:~:text=Bloch's%20Builder%20pattern%20can%20be,for%20a%20missing%20language%20feature.&text=The%20Builder%20pattern%2C%20which%20is,complex%20objects%20step%20by%20step.) also criticizes this duplication - in our case it seems very unlikely that we'd need the Builder for other objects.

Reply:
Please correct me if I'm wrong, but I would like to clarify if you meant something like this:

```java
public static class Builder {
    ...
    private RepoConfiguration repoConfig;

    public Builder() {
        this.repoConfig = new RepoConfiguration();
    }

    ...

    public Builder branch(String branch) {
        // set*() methods should be private
        this.repoConfig.setBranch(branch);
        return this;
    }

    public RepoConfiguration build() {
        return this.repoConfig;
    }
}
```

If so, I believe that this might unintentionally result in aliasing issues. Since the builder pattern allows us to build on existing builder objects to create new configurations or reuse old configurations, the `build` method may unintentionally return the reference to the same `RepoConfiguration` object in memory, regardless of what configurations have been previously applied to the `Builder` object before.

This might violate the builder pattern since the `build` method is supposed to return a new `RepoConfiguration` if the builder configurations are modified, but now it only returns the reference to the latest modified `RepoConfiguration` object, and all previous `RepoConfiguration` references now point to the latest modified `RepoConfiguration` (all previously built `RepoConfiguration` are "forgotten").

Reply:
That's reasonable! I don't think we currently have any builder re-use (since this wouldn't have been possible with the constructors), but it seems possible that we might want to keep a builder for several `RepoConfiguration`s. It also seems like Java's `clone` is a bit of a slippery slope, so it might be unfortunately that this is the best we can do.

Reply:
@gok99 Just a tiny update, I have gone ahead and tried to implement your suggestion by converting the constructor for `RepoConfiguration` into a copy constructor, and created a private empty constructor for `RepoConfiguration` (for use within the `Builder` class).

So far, I managed to remove most of the duplicated variable names within the `Builder` class, spare for the `displayName`, `repoFolderName` and `outputFolderName`, as they are needed for the changing of the names during the actual build step for `Builder`, however, I am looking at implementing flags within `Builder` instead to track if the user has made any modifications to these names, and stop the default action (which is to process the default names for the repo and tag on extra information to it) from taking place.

Reply:
> I am looking at implementing flags within Builder instead to track if the user has made any modifications to these names, and stop the default action (which is to process the default names for the repo and tag on extra information to it) from taking place.

You could probably do the construction of the names within the builder. Null checks are okay, my previous comment on optionals was more to set `displayName` etc to be optionals rather than using them just for the control flow. As long we make sure that the field is eventually set before the object is built, nulls seem fine to use.

> If so, I believe that this might unintentionally result in aliasing issues.

Also regarding aliasing, I realized `RepoConfiguration` instances would still share non-primitive fields which could still cause undesirable aliasing issues. We could possibly reset the RepoConfiguration instance when `build` is called, to avoid this.

Reply:
Regarding the first point, I have mostly converted the null checks into Optional checks, hopefully making the code much more readable and understandable by abstracting away the logic. I have also moved the construction of the name over into the builder class.

Regarding the second point, I have modified the `build` method accordingly and utilised the copy constructor along with the resetting of the built `RepoConfiguration` object within the builder after it is done building. I realise that this might impact the reusability of the builder objects, but as we are not reusing builder objects anywhere as mentioned previously, this should be fine for now.

Reply:
Should we get rid of the copy constructor if we're resetting the object? We could create a separate issue for implementing `Cloneable`s but even if we go that route, we wouldn't need the constructor.

Reply:
Sure, we could stick with the previous builder constructor to enable deep cloning later in the future!